lib/embiggen/uri.rb

```diff
@@ -41,10 +41,22 @@ def follow(request_options = {})
       timeout = request_options.fetch(:timeout) { Configuration.timeout }
 
       location = http_client.follow(timeout)
+      check_location(location)
+
+      location
+    end
+
+    def check_location(location)
       fail BadShortenedURI.new(
         "following #{uri} did not redirect", uri) unless location
 
-      location
+      parsed_uri = Addressable::URI.parse(location)
+      return if parsed_uri.scheme && parsed_uri.host && parsed_uri.path
+
+      fail Addressable::URI::InvalidURIError
```

Comment:
I wonder if these are actually two separate cases and we shouldn't fake an `InvalidURIError` here:

1. The returned location is genuinely an invalid URI (which is surprisingly rare but can be triggered with the URI `::http`);
2. The returned location is a URI but we can't follow it because it's not absolute.

I think we could actually use Addressable's `absolute?` as the check here, e.g.

```ruby
  location = http_client.follow(timeout)
  fail BadShortenedURI.new("cannot follow #{uri}'s redirection", uri) unless followable?(location)

  location
end

def followable?(location)
  return unless location

  Addressable::URI.parse(location).absolute?
rescue Addressable::URI::InvalidURIError => e
  raise BadShortenedURI.new("following #{uri} returns an invalid URI: #{e}", uri)
end
```